source/Octostache.Tests/UsageFixture.cs

```diff
@@ -232,6 +232,28 @@ public void VariableIndexersAreSupported()
             result.Should().Be("MyPackage");
         }
 
+        [Fact]
+        public void NestedIndexersAreSupported()
+        {
+            var result = Evaluate("#{Octopus.Action.Package[containers[0].container].Registry}", new Dictionary<string, string>
+            {
+                { "Octopus.Action.Package[containers[0].container].Registry", "docker.io" },
+            });
+
+            result.Should().Be("docker.io");
+        }
+        
+        [Fact]
+        public void MultipleNestedIndexersAreSupported()
+        {
+            var result = Evaluate("#{Octopus.Action.Package[array[foo].containers[0].container].Registry}", new Dictionary<string, string>
```

Comment:
@zentron I added a test `NestedIndexIterationIsSupported` in `IterationFixture` that iterates the array when the index contains nested `[]`.
I tried to iterate through the inner array but it doesn't seem to work at the moment. Would that be ok for the scope of this change?

Reply:
for consistency It might be worth a timebox to see if it involves much work, but given it didn't do anything at all before its probably reasonable as is.

Reply:
I added a card to our board for this investigation :) https://trello.com/c/ffTk2rHv/67-octostache-investigate-capability-to-iterate-over-nested-arrays-inside-indexes